opentelemetry-api/src/opentelemetry/context/propagation/httptextformat.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import typing
+from opentelemetry.trace import SpanContext
+
+Setter = typing.Callable[[object, str, str], None]
+Getter = typing.Callable[[object, str], str]
+
+
+class HTTPTextFormat(abc.ABC):
+    """API for propagation of spans via headers.
+
+    This class provides an interface that enables extracting and injecting
+    trace state into headers of HTTP requests. Http frameworks and client
+    can integrate with HTTPTextFormat by providing the object containing the
+    headers, and a getter and setter function for the extraction and
+    injection of values, respectively.
+
+    Example::
+
+        import flask
+        import requests
+        from opentelemetry.context.propagation import HTTPTextFormat
+
+        PROPAGATOR = HTTPTextFormat()
```

Comment:
Since `HTTPTextFormat` is the abstract base class that cannot be instantiated, I am guessing this example line is not to be used literally like this, but users would write something like
```
PROPAGATOR = B3Format()
```
Is it correct?

How should users of opentelemetry-python decide which implementation of the base class to pick? Or, should the choice be made for them? It's not clear to me if the spec ([api-propagators.md](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-propagators.md)) mandates a format (B3 / W3C Trace Context) or not.

Reply:
According to spec now they'd do [`Tracer.get_binary_format` (java link)](https://github.com/c24t/opentelemetry-java/blob/b47c0a02e5c0315beb96be88ec3f8c7fdf64b4e5/api/src/main/java/io/opentelemetry/trace/Tracer.java#L213) or `get_text_format`, and the particular binary/text propagator would be determined by config, with the default implementations written in the API package.

Which raises the question: what if an application needs to handle multiple wire formats? Do we need multiple tracers, and do we need the option to set the propagator class at runtime? There may be some discussion in the spec repo for this already, but I don't see a single good source for it.

Reply:
Yes, that is a good question. I could easily imagine that would be helpful, if not necessary, as people migrate from one tracing spec to another.

I also feel like the Tracer API is a weird place to put the propagator getters. It might as well belong in it's own registry. Should I file an RFC around that?

Reply:
Will leave it for now (although is invalid) because we're missing the story on how to populate propagators. I'll work that out an fix this comment in a different PR.